api/api.go

```diff
@@ -501,32 +501,40 @@ func (b *BlockChainAPI) EstimateGas(
 		data = *args.Data
 	} else if args.Input != nil {
 		data = *args.Input
+	} else {
+		return hexutil.Uint64(0), errors.Join(errs.ErrInvalid, fmt.Errorf("either data or input fields has to be provided"))
 	}
 
-	eoa := newEOATestAccount(b.config.COAKey.String()[2:])
-	// Provide a high enough gas for the tx to be able to execute
-	gas := uint64(15_000_000)
+	// provide a high enough gas for the tx to be able to execute
+	defaultGasLimit := uint64(15_000_000)
 	if args.Gas != nil {
-		gas = uint64(*args.Gas)
+		defaultGasLimit = uint64(*args.Gas)
```

Comment:
Nice :ok_hand:

Reply:
yeah if thins don't work better to return some default